src/llfs/track_fds.cpp

```diff
@@ -0,0 +1,188 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/track_fds.hpp>
+//
+
+#include <llfs/logging.hpp>
+
+#include <batteries/env.hpp>
+#include <batteries/suppress.hpp>
+
+#include <filesystem>
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+
+#ifdef __linux__
+//
+#include <dirent.h>
+#include <sys/types.h>
+//
+#endif
+
+namespace llfs {
+
+namespace {
+
+const char* kVarName = "LLFS_TRACK_FDS";
+
+struct TrackFdState {
+  std::atomic<bool> enabled{batt::getenv_as<int>(kVarName).value_or(0) != 0};
```

Comment:
Better to use batt::Watch here?

Reply:
`batt::Watch<bool>` uses `std::atomic<bool>` internally; it would be good to use `Watch` instead when we have a use case where we might need to wait on the value changing.  But there isn't such a use case right now, at least as far as I'm aware.